include/extractor/guidance/intersection.hpp

```diff
@@ -48,13 +49,23 @@ struct TurnOperation final
 // aaaaaaaa
 //
 // We would perceive a->c as a sharp turn, a->b as a slight turn, and b->c as a slight turn.
-struct ConnectedRoad final
+struct ConnectedRoad final : public TurnOperation
 {
+    using Base = TurnOperation;
+
     ConnectedRoad(const TurnOperation turn, const bool entry_allowed = false);
 
     // a turn may be relevant to good instructions, even if we cannot enter the road
     bool entry_allowed;
-    TurnOperation turn;
+
+    // used to sort the set of connected roads (we require sorting throughout turn handling)
+    bool compareByAngle(const ConnectedRoad &other) const;
+
+    // make a left turn into an equivalent right turn and vice versa
+    void mirror();
+
+    OSRM_ATTR_WARN_UNUSED
+    ConnectedRoad getMirroredCopy() const;
```

Comment:
At the moment a ConnectedRoad does not know if it is the UTurn road or not - this is implicit in the parent Intersection ordering. Should there be a boolean flag here indicating this instead? Otherwise we always need the Intersection object when ConnectedRoads (think first, last iterators) would be good enough.